sycl/test-e2e/ESIMD/unified_memory_api/gather_acc_dg2_pvc.cpp

```diff
@@ -0,0 +1,43 @@
+//==------- gather_acc_dg2_pvc.cpp - DPC++ ESIMD on-device test ------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: gpu-intel-dg2 || gpu-intel-pvc
+
+// Use per-kernel compilation to have more information about failing cases.
+// RUN: %{build} -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %{run} %t.out
+
+// The test verifies esimd::gather() functions accepting ACCESSOR
+// and optional compile-time esimd::properties.
+// The gather() calls in this test do not use cache-hint properties
+// or VS > 1 (number of loads per offset) to not impose using PVC features.
+
+#include "Inputs/gather.hpp"
+
+int main() {
+  auto Q = queue{gpu_selector_v};
+  esimd_test::printTestLabel(Q);
+
+  constexpr auto TestFeatures = TestFeatures::DG2;
+  bool Passed = true;
+
+  Passed &= testACC<int8_t, TestFeatures>(Q);
+  Passed &= testACC<int16_t, TestFeatures>(Q);
+  if (Q.get_device().has(sycl::aspect::fp16))
+    Passed &= testACC<sycl::half, TestFeatures>(Q);
+  Passed &= testACC<uint32_t, TestFeatures>(Q);
+  Passed &= testACC<float, TestFeatures>(Q);
+  Passed &= testACC<ext::intel::experimental::esimd::tfloat32, TestFeatures>(Q);
+#ifdef __ESIMD_FORCE_STATELESS_MEM
```

Comment:
Should we be running this test and the previous in stateless mode as well? It seems we are not right now.

Reply:
Yes, good catch. I'll add the tests here soon.